src/Enghouse/client/indexWatsonOptions.html

```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
```

Comment:
There is an awful lot of console calls in your code. I think it would be a good idea to put everything that is not a warning or an error under a debug flag that a customer can turn on if they wish. Adding another constructor argument would be a good way to do that. Otherwise this is very noisy and if a customer is trying to debug their own site, having a massive amount of output from your script might make that difficult. It is particularly noisy because it is sending out a constant stream of messages while a user is connected to an agent.

Reply:
> Unfortunately, there are still some problems with how you are handling the initialization of web chat. You can't actually rely on `window.watsonAssistantChatOptions` to exist. Not all instances of web chat use `window.watsonAssistantChatOptions` because that object is not actually required. Our react library uses an alternative mechanism for loading web chat that will not work with your script. I think all references to `watsonAssistantChatOptions` should be removed from your script.
> 
> This also means removing the `setTimeout` that injects the web chat script. Some customers actually use a proxy to access web chat and don't allow direct internet access. Hardcoding the URL to web chat in your script would not work for those customers.
> 
> Also, you are overriding the `onLoad` function which means any customizations the customer may need to perform after web chat is loaded is not possible (like registering event listeners or customizing theme colors). I think that is going to be a blocker because a lot of customers need to do that. You should do what I recommended in my previous note and export your SD class and let the customer inject it into their script. Because your class also needs access to the web chat instance, you'll need to tweak how the factory would be used but I'll email you an example.

I am only familiar with scenario to use the `window.watsonAssistantChatOptions` to load the service desk etc. - how can I test without this?

Reply:
We haven't documented it but it basically amounts to using `window.loadWatsonAssistantChat(options)` where `options` can be any object; it doesn't have to be `window.watsonAssistantChatOptions`. When using `loadWatsonAssistantChat`, you do not use `onLoad` in the options object. Instead, you await the result and that's how you get the instance. Here's an example.
```
<script>
  const options = {
    integrationID: "...",
    region: "...",
    serviceInstanceID: "...",
    subscriptionID: "...",
  };

  async function loadWebChat() {
    const instance = await window.loadWatsonAssistantChat(options);
    instance.render();
  }

  setTimeout(function(){
    const t=document.createElement('script');
    t.src='https://web-chat.global.assistant.watson.appdomain.cloud/versions/' + (options.clientVersion || 'latest') + '/WatsonAssistantChatEntry.js';
    t.onload = loadWebChat;
    document.head.appendChild(t);
  });
</script>
```

Reply:
so all is fine - we exactly implemented it in that way. You can check recent indexWatson.html source